src/main/java/core/basesyntax/service/impl/ReaderImpl.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitsTransactions;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Reader;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class ReaderImpl implements Reader {
+    private static final String DELIMITER = ",";
+
+    @Override
+    public void readerTransaction(String fileName) {
+        try (BufferedReader bufferedReader = new BufferedReader((new FileReader(fileName)))) {
+            String text = bufferedReader.readLine();
+            text = bufferedReader.readLine();
+            while (text != null) {
+                addFruitTransaction(text);
+                text = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can`open file" + fileName, e);
+        }
+    }
+
```

Comment:
You must separate different logic. In ReaderImpl you must only read from file and save to list, and return this list. Non process data.